### PR TITLE
Add ability to differentiate timeout errors

### PR DESCRIPTION
**Pull request by jmhooper**
**Why**: As part of LG-650 we are handling timeouts and making the IDP
aware of them. This commit adds the ability for proofer subclasses to
raise a specific error during a timeout which will be rescued and
reported to the app.

### CODE REVIEW

**Comment by jmhooper** [comment]
I opened #47 to upgrade reek and fix the CodeClimate issue

**Comment by jmhooper** [comment]
#47 is merged and I've merged those changes into this so now CC is green.

**Comment by jmhooper** [comment]
There is a pending snyk check that is never going to complete because this repo doesn't have a Gemfile.lock

File: lib/proofer/result.rb [hunk 1]
```diff
@@ -47,6 +47,10 @@ def success?
       !exception? && !errors?
     end
 
+    def timed_out?
+      exception? && @exception.is_a?(Proofer::TimeoutError)
```
Comment (jgsmith-usds) [inline lib/proofer/result.rb]: Style: `@exception&.is_a?(Proofer::TimeoutError)`?

**Review by jgsmith-usds — APPROVED** [review]
LGTM

File: lib/proofer/result.rb (hunk 1 above)
Comment (jmhooper) [reply lib/proofer/result.rb]: I actually don't think we need to guard against nil here b/c `Nilclass` is not an instance of `Proofer::TimeouError`. I'll tie that up before merging.

**Pull request merged**